packages/psss/src/store/auth.js

```diff
@@ -5,69 +5,48 @@
 
 import { createSelector } from 'reselect';
 import { AccessPolicy } from '@tupaia/access-policy';
+import { loginUser, logoutUser, updateUser, getUser } from '../api';
 import { createReducer } from '../utils/createReducer';
 
 // actions
-const LOGIN_START = 'LOGIN_START';
 const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
-const LOGIN_ERROR = 'LOGIN_ERROR';
 const LOGOUT = 'LOGOUT';
-export const PROFILE_SUCCESS = 'PROFILE_SUCCESS';
+const PROFILE_SUCCESS = 'PROFILE_SUCCESS';
 
 // action creators
-export const login = ({ email, password }) => async (dispatch, getState, { api }) => {
+export const login = ({ email, password }) => async dispatch => {
   const deviceName = window.navigator.userAgent;
-
-  dispatch({ type: LOGIN_START });
-  try {
-    const userDetails = await api.reauthenticate({
-      emailAddress: email,
-      password,
-      deviceName,
-    });
-    dispatch(loginSuccess(userDetails));
-  } catch (error) {
-    dispatch(loginError(error.message));
-  }
+  const user = await loginUser({
```

Comment:
How does the error handling work here?

Reply:
(or in the others below)

Reply:
The error will bubble up and get handled in the try catch of the component (eg. LoginFormComponent). This does create a risk that if someone in the future dispatched login without a try catch in a component then there would be a risk of an uncaught exception. If the errors were handled in the store code then, the same risk would just exist in a different place.

Reply:
I did a little bit of extra research around handling unchaught exceptions. The good news is that for all the queries that use react-query (ie. all but the 3 in this file). Then there is no risk because, react-query handles the thrown error and simply returns the error object with the query response.